### PR TITLE
Snapshot testing + current results snapshot

### PR DESCRIPTION
**Pull request by kosciej**
Snapshot testing takes advantage of deterministic character of Cleora. Any discrepancies between original snapshot results and current ones can be then reviewed along with the code which introduced discrepancy.
Test introduced by this PR performs work for sample case and saves snapshot file.

### CODE REVIEW

**Comment by kosciej** [comment]
It works correctly on:
stable-x86_64-apple-darwin
stable-x86_64-unknown-linux-gnu

for stable-armv7-unknown-linux-gnueabihf (ARM RaspberryPi), the order of HashMaps is different in debug string, and therefore snapshots didn't check for SparseMatrices. It works correctly for final result.

To prevent this kind of issue I can drop collecting snapshots for intermediate SnapshotMatrices or work out another solution to make snapshots stable.

File: tests/snapshot.rs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        config.max_number_of_iteration,
+        in_memory_entity_mapping_persistor,
+        &mut in_memory_embedding_persistor,
+    );
+
+    assert_debug_snapshot!("embeddings", in_memory_embedding_persistor);
+}
+
+#[derive(Debug, Default)]
+pub struct InMemoryEmbeddingPersistor {
```
Comment (piobab) [inline tests/snapshot.rs:73]: I think you can remove `pub` from `InMemoryEmbeddingPersistor` and `InMemoryEntity` structs.

Comment (kosciej) [reply tests/snapshot.rs]: done

**Comment by kosciej** [comment]
Removed snapshot for sparse matrices, as it's intermediate result that contains hash table, which under some circumstances have different ordering (affects snapshot comparison).

**Review by piobab — APPROVED** [review]

**Pull request merged**